### PR TITLE
Add workaround so ItemsRepeater doesn't leak the result of DataTemplate::Load

### LINKED ISSUE #1954: Is ItemsRepeater leaky
**Describe the bug**
@ranjeshj 
When navigating away from a page containing `ItemsRepeater`, `ItemsRepeater` does not release the all the Item views.

**Steps to reproduce the bug**
Please get the repo app from [here](https://github.com/harvinders/ItemsRepeaterAnimationApp)

1. Click on a image, it takes you to next page
2. Click back button
3. Repeat [1] and [2] multiple times
4. End with the clicking on the Image
5. Take a memory snapshot and check the `StoryView` count, 

**Expected behavior**
ItemsRepeater should not leak

**Version Info**

NuGet package version: 2.4.0-prerelease.200203002

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (xxxxx)              |    |
| November 2019 Update (18363)  |    |
| May 2019 Update (18362)            |    |
| October 2018 Update (17763)        | Yes|
| April 2018 Update (17134)          |    |
| Fall Creators Update (16299)       |    |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :-------------------- | :------------------- |
| Desktop                 | Yes  |
| Mobile                   |    |
| Xbox                      |    |
| Surface Hub          |     |
| IoT                         |     |

**Additional context**

### PR DESCRIPTION
**Pull request by jevansaks**
The thread of #1954 has lots more information, but the short story is that ItemsRepeater has a call to DataTemplate::LoadContent to decide if the template is empty. DataTemplate::LoadContent has a bug (#3057) where if you don't add the result to a UIElementCollection then the returned object will be leaked. The simplest workaround is to just add the result to the ItemsRepeater's own child collection and immediately remove it.

Thanks @chingucoding for the test code. I changed the scenario a little from what you had built because I wanted to verify that all the items get cleaned up.

Fixes #1954

### CODE REVIEW

File: dev/Repeater/TestUI/Samples/ItemTemplateSamples/DisposableUserControl.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+using Windows.UI.Xaml.Navigation;
+
+// The User Control item template is documented at https://go.microsoft.com/fwlink/?LinkId=234236
+
+namespace MUXControlsTestApp.Samples
+{
+    public sealed partial class DisposableUserControl : UserControl
+    {
+
+        private int num;
```
Comment (ranjeshj) [inline dev/Repeater/TestUI/Samples/ItemTemplateSamples/DisposableUserControl.xaml.cs:24]: why is this needed ?

Comment (jevansaks) [reply dev/Repeater/TestUI/Samples/ItemTemplateSamples/DisposableUserControl.xaml.cs:24]: That was from @chingucoding's branch. Looks like it's not really needed since we now understand that custom properties don't come into play. I can remove the unnecessary bits.

Comment (ranjeshj) [reply dev/Repeater/TestUI/Samples/ItemTemplateSamples/DisposableUserControl.xaml.cs:24]: I'm guessing we dont need the custom user control at all ?

Comment (jevansaks) [reply dev/Repeater/TestUI/Samples/ItemTemplateSamples/DisposableUserControl.xaml.cs:24]: We need something to be able to track that we leaked one of the objects. So either a custom user control or custom property would do it. The custom UserControl was what started the discussion so I like that as a repro scenario.

File: dev/Repeater/APITests/RepeaterTests.cs [hunk 2]
```diff
@@ -609,5 +612,62 @@ public void VerifyUIElementsInItemsSource()
...
+                Verify.IsGreaterThanOrEqual(MUXControlsTestApp.Samples.DisposableUserControl.OpenItems, 10, "Verify we created at least 10 DisposableUserControl");
+
+                // Clear out the repeater and make sure everything gets cleaned up.
+                Content = null;
+                repeater = null;
+            });
+
+            IdleSynchronizer.Wait();
+            // Wait until the Tick to make sure all layout has completed.
+            RunOnUIThread.WaitForTick();
```
Comment (chingucoding) [inline dev/Repeater/APITests/RepeaterTests.cs:663]: I am not entirely sure if we need this, on my machine the test worked fine without it, but maybe this is depending on different factors, such as CPU usage etc.

File: dev/Repeater/APITests/RepeaterTests.cs [hunk 3]
```diff
@@ -609,5 +612,62 @@ public void VerifyUIElementsInItemsSource()
                 }
             });
         }
+    
+        [TestMethod]
+        public void VerifyRepeaterDoesNotLeakItemContainers()
```
Comment (chingucoding) [inline dev/Repeater/APITests/RepeaterTests.cs:617]: Great idea to check that we actually clear all of the elements! That makes a lot more sense than verifying we only have 10 open at the end of update layout.

File: dev/Repeater/TestUI/Samples/ItemTemplateSamples/DisposableUserControl.xaml.cs (hunk 1 above)
Comment (ranjeshj) [reply dev/Repeater/TestUI/Samples/ItemTemplateSamples/DisposableUserControl.xaml.cs]: Right, we need something to track, but don't need it to be a user control with markup. I was thinking something like below but both work fine. 
```
class LeakTrackingContainer: ContentControl
{
    ctor(){}
    dtor{}
    static .. instanceCount
    bool IsLeaking()
    {
        return instanceCount != 0
    }
}
```

File: test/MUXControlsTestApp/Utilities/RunOnUIThread.cs [hunk 4]
```diff
@@ -90,19 +90,26 @@ public static void Execute(CoreApplicationView whichView, Action action)
...
             {
-                renderingEventFired.TrySetResult(null);
+                renderingEvent.Set();
             };
-            CompositionTarget.Rendering += renderingCallback;
 
-            await renderingEventFired.Task;
+            RunOnUIThread.Execute(() =>
+            {
+                Windows.UI.Xaml.Media.CompositionTarget.Rendering += renderingHandler;
```
[the remark this answers is not in the document]
Comment (jevansaks) [reply test/MUXControlsTestApp/Utilities/RunOnUIThread.cs:104]: Fixed.

File: dev/Repeater/TestUI/Samples/ItemTemplateSamples/ItemTemplateDemo.xaml [hunk 5]
```diff
@@ -99,5 +100,21 @@
...
+                                            Layout="{StaticResource stackLayout}">
+                        <controls:ItemsRepeater.ItemTemplate>
+                            <DataTemplate x:DataType="local:MyData" >
+                                <local:DisposableUserControl Number="{x:Bind number}"/>
+                            </DataTemplate>
+                        </controls:ItemsRepeater.ItemTemplate>
+                    </controls:ItemsRepeater>
+                </ScrollViewer>
+            </controls:ItemsRepeaterScrollHost>
+        </StackPanel>
```
[the remark this answers is not in the document]
Comment (jevansaks) [reply dev/Repeater/TestUI/Samples/ItemTemplateSamples/ItemTemplateDemo.xaml:118]: I still found it useful for manual testing. But there is certainly a meta issue of duplication between manual tests and API tests (which i won't be resolving here :) ).

File: dev/Repeater/APITests/RepeaterTests.cs (hunk 2 above)
Comment (jevansaks) [reply dev/Repeater/APITests/RepeaterTests.cs:663]: You might be right, i'll try removing it.

File: dev/Repeater/TestUI/Samples/ItemTemplateSamples/DisposableUserControl.xaml.cs (hunk 1 above)
Comment (jevansaks) [reply dev/Repeater/TestUI/Samples/ItemTemplateSamples/DisposableUserControl.xaml.cs]: Sure, that would work too, but i'll keep it as-is.
